optimum/habana/accelerate/data_loader.py

```diff
@@ -1,451 +0,0 @@
-            - `"generator"`: the `torch.Generator` of the sampler (or batch sampler if there is no sampler in your
-              dataloader) or of the iterable dataset (if it exists) if the underlying dataset is of that type.
-
-        dispatch_batches (`bool`, *optional*):
-            If set to `True`, the datalaoder prepared is only iterated through on the main process and then the batches
-            are split and broadcast to each process. Will default to `True` when the underlying dataset is an
-            `IterableDataset`, `False` otherwise.
-        even_batches (`bool`, *optional*, defaults to `True`):
-            If set to `True`, in cases where the total batch size across all processes does not exactly divide the
-            dataset, samples at the start of the dataset will be duplicated so the batch can be divided equally among
-            all workers.
-        slice_fn_for_dispatch (`Callable`, *optional*`):
-            If passed, this function will be used to slice tensors across `num_processes`. Will default to
-            [`~utils.slice_tensors`]. This argument is used only when `dispatch_batches` is set to `True` and will be
-            ignored otherwise.
-        use_seedable_sampler (`bool`, *optional*, defaults to `False`):
-            Whether to use the [`~data_loader.SeedableRandomSampler`] instead of a `RandomSampler` for better
-            reproducability. Comes at a cost of potentially different performances due to different shuffling
-            algorithms but ensures results will be the *exact* same. Should be paired with `set_seed()` at every
-            `self.set_epoch`
-        non_blocking (`bool`, *optional*, defaults to `False`):
-            If set to `True`, dataloader will utilize non-blocking host-to-device transfers. If the dataloader has
-            `pin_memory` set to `True`, this will help to increase overlap between data transfer and computations.
-
-    Returns:
-        `torch.utils.data.dataloader.DataLoader`: A new data loader that will yield the portion of the batches
-
-    <Tip warning={true}>
-
-    `BatchSampler`s with varying batch sizes are not enabled by default. To enable this behaviour, set `even_batches`
-    equal to `False`
-
-    </Tip>
-    """
-    if dispatch_batches is None:
-        if not put_on_device:
-            dispatch_batches = False
-        else:
-            dispatch_batches = isinstance(dataloader.dataset, IterableDataset)
-
-    if dispatch_batches and not put_on_device:
-        raise ValueError("Using `dispatch_batches=True` requires `put_on_device=True`.")
-    # Grab defaults from GaudiAcceleratorState
-    state = GaudiAcceleratorState()
-    if num_processes is None:
-        num_processes = state.num_processes
-    if process_index is None:
-        process_index = state.process_index
-    if parallel_state.sequence_parallel_is_initialized() and parallel_state.get_sequence_parallel_world_size() > 1:
```

Comment:
@IlyasMoutawwakil the `parallel_state` is also a OH patch based on megatron and not available on upstream

Reply:
This was simply moved to the accelerator https://github.com/huggingface/optimum-habana/pull/1876/files#diff-a12fd0307be36a45df28c68e5ccb1eea4aba070731173c51437aea0b2c1c4496R658

Reply:
I am not sure it is possible for this to be moved to accelerator! this is only available in OH. In accelerator, this only exists when using megatron so it does not work for OH cases

Reply:
sequence_parallel in OH is one of patches adapting from megatron to enable long seq processing without actually using megatron

Reply:
please see the code I tagged you in.